reactivesocket-tck-drivers/src/main/java/io/reactivesocket/tckdrivers/client/JavaClientDriver.java

```diff
@@ -0,0 +1,554 @@
+                    break;
+                case "channel":
+                    handleChannel(args, iter, name);
+                    break;
+                case "echochannel":
+                    handleEchoChannel(args);
+                    break;
+                case "await":
+                    switch (args[1]) {
+                        case "terminal":
+                            handleAwaitTerminal(args);
+                            break;
+                        case "atLeast":
+                            handleAwaitAtLeast(args);
+                            break;
+                        case "no_events":
+                            handleAwaitNoEvents(args);
+                            break;
+                        default:
+                            break;
+                    }
+                    break;
+
+                case "assert":
+                    switch (args[1]) {
+                        case "no_error":
+                            handleNoError(args);
+                            break;
+                        case "error":
+                            handleError(args);
+                            break;
+                        case "received":
+                            handleReceived(args);
+                            break;
+                        case "received_n":
+                            handleReceivedN(args);
+                            break;
+                        case "received_at_least":
+                            handleReceivedAtLeast(args);
+                            break;
+                        case "completed":
+                            handleCompleted(args);
+                            break;
+                        case "no_completed":
+                            handleNoCompleted(args);
+                            break;
+                        case "canceled":
+                            handleCancelled(args);
+                            break;
+                    }
+                    break;
+                case "take":
+                    handleTake(args);
+                    break;
+                case "request":
+                    handleRequest(args);
+                    break;
+                case "cancel":
+                    handleCancel(args);
+                    break;
+                case "EOF":
+
+                    break;
+                default:
+
+                    break;
+            }
+
+        }
+        // this check each of the subscribers to see that they all passed their assertions
+        if (id.size() > 0) {
+            boolean hasPassed = true;
+            for (String str : id) {
+                if (payloadSubscribers.get(str) != null) hasPassed = hasPassed && payloadSubscribers.get(str).hasPassed();
+                else hasPassed = hasPassed && fnfSubscribers.get(str).hasPassed();
+            }
+            return Optional.of(hasPassed);
+        }
+        else return Optional.empty();
```

Comment:
Maybe a better return type would be `boolean`, and use an exception for when `id.size() == 0`